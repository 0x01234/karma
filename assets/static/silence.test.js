const $ = window.jQuery = require("jquery");
const templatesMock = require("./__mocks__/templatesMock");
const ajaxMock = require("./__mocks__/ajaxMock");

test("silence setupSilenceForm()", () => {
    var body = templatesMock.loadTemplates();
    document.body.innerHTML = body;

    const silence = require("./silence");
    silence.setupSilenceForm();
});

test("successful sendSilencePOST()", () => {
    var body = templatesMock.loadTemplates();
    body.push(
        "<span class='silence-post-result' " +
        "      data-uri='http://localhost'>" +
        "</span>"
    );
    document.body.innerHTML = body;

    const templates = require("./templates");
    templates.init();

    const ajaxServer = ajaxMock.createServer(200, {
        "status": "success",
        "data": {"silenceId": "abcdef"}
    });
    ajaxServer.start();

    const silence = require("./silence");
    silence.sendSilencePOST("http://localhost", {});

    let resultElem = $(".silence-post-result").html().trim();
    expect(resultElem).toMatchSnapshot();

    ajaxServer.stop();
});

test("failed sendSilencePOST()", () => {
    var body = templatesMock.loadTemplates();
    body.push(
        "<span class='silence-post-result' " +
        "      data-uri='http://localhost'>" +
        "</span>"
    );
    document.body.innerHTML = body;

    const templates = require("./templates");
    templates.init();

    const ajaxServer = ajaxMock.createServer(500, {
        "status": "success",
        "errorType": "server_error",
        "error": "request failed"
    });
    ajaxServer.start();

    const silence = require("./silence");
    silence.sendSilencePOST("http://localhost", {});

    let resultElem = $(".silence-post-result").html().trim();
    expect(resultElem).toMatchSnapshot();

    ajaxServer.stop();
});
