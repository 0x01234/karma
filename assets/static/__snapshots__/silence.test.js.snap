// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`failed sendSilencePOST() 1`] = `
"<p class=\\"text-danger\\">
  <i class=\\"fa fa-exclamation-circle\\"></i>
  request failed
</p>"
`;

exports[`successful sendSilencePOST() 1`] = `
"<p class=\\"text-success\\">
  <i class=\\"fa fa-check-circle\\"></i>
  abcdef
</p>"
`;
